large-message-core/src/main/java/com/bakdata/kafka/AmazonS3Client.java

```diff
@@ -70,43 +88,49 @@ public void deleteAllObjects(final String bucket, final String prefix) {
-        try (final InputStream content = new ByteArrayInputStream(bytes)) {
-            final ObjectMetadata metadata = createMetadata(bytes);
-            this.s3.putObject(bucket, key, content, metadata);
+        try {
+            final PutObjectRequest request = PutObjectRequest.builder()
+                    .bucket(bucket)
+                    .key(key)
+                    .build();
+            this.s3.putObject(request, RequestBody.fromBytes(bytes));
             return asURI(bucket, key);
-        } catch (final IOException e) {
+        } catch (final SdkException e) {
             throw new SerializationException("Error backing message on S3", e);
         }
     }
 
     @Override
     public byte[] getObject(final String bucket, final String key) {
         final String s3URI = asURI(bucket, key);
-        try (final S3Object s3Object = this.s3.getObject(bucket, key);
-                final InputStream in = s3Object.getObjectContent()) {
-            return IOUtils.toByteArray(in);
-        } catch (final IOException e) {
+        try (final ResponseInputStream<GetObjectResponse> s3Object = this.s3.getObject(
+                GetObjectRequest.builder().bucket(bucket).key(key).build())) {
+            return IoUtils.toByteArray(s3Object);
+        } catch (final SdkException | IOException e) {
             throw new SerializationException("Cannot handle S3 backed message: " + s3URI, e);
         }
     }
 
     private void deleteObjects(final String bucketName, final String prefix) {
-        ObjectListing objectListing = this.s3.listObjects(bucketName, prefix);
+        final Builder requestBuilder = ListObjectsRequest.builder()
+                .bucket(bucketName)
+                .prefix(prefix);
+        ListObjectsResponse objectListing = this.s3.listObjects(requestBuilder.build());
         while (true) {
-            final String[] keys = objectListing.getObjectSummaries().stream()
-                    .map(S3ObjectSummary::getKey)
-                    .toArray(String[]::new);
-            if (keys.length > 0) {
-                this.s3.deleteObjects(new DeleteObjectsRequest(bucketName)
-                        .withKeys(keys));
-            }
+            final List<ObjectIdentifier> keys = objectListing.contents().stream()
+                    .map(AmazonS3Client::asIdentifier)
+                    .collect(Collectors.toList());
+            this.delete(bucketName, keys);
 
             // If the bucket contains many objects, the listObjects() call
             // might not return all of the objects in the first listing. Check to
             // see whether the listing was truncated. If so, retrieve the next page of objects
             // and delete them.
-            if (objectListing.isTruncated()) {
-                objectListing = this.s3.listNextBatchOfObjects(objectListing);
+            if (Boolean.TRUE.equals(objectListing.isTruncated())) {
```

Comment:
Do you know if it ever returns null? Or is it just a precaution? Because I wouldn't know what null means in this context

Reply:
isTruncated returns a `Boolean` and sonarqube complained so I just changed it